### PR TITLE
Create home page API

### LINKED ISSUE #6: API: Create API definition for Home Page
Based on this suggestion: 

```
"opportunities_and_programmes": {
    "title": "Opportunities and Programmes",
    "description": "Join our community and unlock endless opportunities. Network, find mentors, and access leadership programs. Whether you're aiming to enhance your skills, grow your professional network, or advance your career, we have what you need. We offer a wide range of opportunities to help you achieve your goals.",
    "programmes": [
      { "name": "Mentorship", "link": "#" },
      { "name": "Online and in-person Events", "link": "#" },
      { "name": "Book club", "link": "#" },
      { "name": "CV clinic", "link": "#" },
      { "name": "Mock interviews", "link": "#" },
      { "name": "Leetcode", "link": "#" }
    ]
  },
  "mentorship_ad_hoc": {
    "title": "Mentorship ad-hoc registration is open now",
    "registration_dates": "Registration: MAY 6 - MAY 17, 2024", // could be send with start/end, whatever suits best
    "description": "Sign up for this unique opportunity of one-time mentoring sessions with personalised guidance from experienced mentors.",
    "cta": { "text": "Register now", "link": "#" },
    "image": "mentorship_ad_hoc_image.jpg"
  },
  "events_section": {
    "title": "Events",
    "view_all_events_link": "#",
    "events": [
      {
        "type": "Book Club",
        "date_time": "THU, MAY 30, 2024, 8:00 PM to 9:30 PM CEST",
        "title": "Book Club: The Pragmatic Programmer",
        "hosts": "Sasha Northon and Lana Rey",
        "description": "Join us for a discussion of this essential guide to writing clear and maintainable code. We recommend that our attendees read the book before the event, however you are welcome to join us, even if you have only read the first chapter. The more, the merrier!",
        "image": "the_pragmatic_programmer.jpg",
        "cta": { "text": "Go to Meetup Event", "link": "#" }
      },
      {
        "type": "Tech Talk",
        "date_time": "THU, JUN 6, 2024, 8:00 PM to 9:30 PM CEST",
        "title": "Getting Started with IaC - An Intro to Terraform",
        "speaker": "Anna Lee, Lead Engineer",
        "description": "Join Women Coding Community for an introductory session on Infrastructure-as-code (IaC) and Terraform with speaker Anna Lee, Senior/Lead Engineer. Whether you're a seasoned developer, a systems administrator, or just curious about cloud infrastructure, this talk is designed to equip you with the knowledge to start using Terraform effectively.",
        "image": "anna_lee.jpg",
        "cta": { "text": "Go to Meetup Event", "link": "#" }
      }
    ]
  },
  "volunteer_section": {
    "title": "Volunteer With Us",
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit.",
    "cta": { "text": "Learn more about volunteering", "link": "#" },
    "image": "volunteer_image.jpg"
  }
}
```

### PR DESCRIPTION
**Pull request by dricazenck**
Closes #6 

image
image

### CODE REVIEW

**Comment by mads6879** [comment]
@dricazenck - Just a thought - Instead of making sections static, would it be better if we keep them variable so it can be expanded it future easily? 
I don't know how the frontend team plans to use the response but if we return sections as array of objects then they can simply loop through the sections array and don't really need to know each name of the section (for the object key e.g. **HomeSection**, **ProgrammesSection**).

**tag** can be camelCase or snake_case as per the convention u r following.

`{
    "title" : "Home page",
    "tag" : "home_page",
    "description": "Description of the page",
    "totalSections": 1,
    "sections" : [
        {
            "title" : "Home section",
            "tag" : "home_section",
            "description" : "Description of the section",
            "images": [
                {
                    "path": "something",
                    "alt": "something",
                    "type": "desktop"
                }
            ]
        }
    ]
}`

image

**Comment by dricazenck** [comment]
> @dricazenck - Just a thought - Instead of making sections static, would it be better if we keep them variable so it can be expanded it future easily? I don't know how the frontend team plans to use the response but if we return sections as array of objects then they can simply loop through the sections array and don't really need to know each name of the section (for the object key e.g. **HomeSection**, **ProgrammesSection**).
> 
> **tag** can be camelCase or snake_case as per the convention u r following.
> 
> `{ "title" : "Home page", "tag" : "home_page", "description": "Description of the page", "totalSections": 1, "sections" : [ { "title" : "Home section", "tag" : "home_section", "description" : "Description of the section", "images": [ { "path": "something", "alt": "something", "type": "desktop" } ] } ] }`
> 
> image

@mads6879 that was a suggestion from the frontend team. Actually for me looks like the same approach just using different attributes names. We discussed to create sections and the frontend will create the same config to correlate with the React components to render the content. 

In the future, when we create the admin, we will have a easy way to associate new pages to components.

**Comment by stepsen89** [comment]
> @dricazenck - Just a thought - Instead of making sections static, would it be better if we keep them variable so it can be expanded it future easily? I don't know how the frontend team plans to use the response but if we return sections as array of objects then they can simply loop through the sections array and don't really need to know each name of the section (for the object key e.g. **HomeSection**, **ProgrammesSection**).
> 
> **tag** can be camelCase or snake_case as per the convention u r following.
> 
> `{ "title" : "Home page", "tag" : "home_page", "description": "Description of the page", "totalSections": 1, "sections" : [ { "title" : "Home section", "tag" : "home_section", "description" : "Description of the section", "images": [ { "path": "something", "alt": "something", "type": "desktop" } ] } ] }`
> 
> image

Not sure if I get what you mean, but the FE needs to know the name so we apply the correct components/setup for it 🤔 not all sections are the same, and if we add more the FE also needs to add that specific section/setup

File: resources/home.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    ]
+  },
+  "programmesSection": {
+    "title": "Opportunities and Programmes",
+    "description": "Join our community and unlock endless opportunities. Network, find mentors, and access leadership programs. Whether you're aiming to enhance your skills, grow your professional network, or advance your career, we have what you need. We offer a wide range of opportunities to help you achieve your goals.\n",
+    "programmes": [
+      {
+        "name": "Mentorship",
+        "link": "/mentorship",
+        "icon": "Diversity2"
```
Comment (stepsen89) [inline resources/home.json:20]: which icon set are you referring too? we would need to agree how/what icons we need - there are some which needs to be imported, but others where the exact name would be enough

**Comment by mads6879** [comment]
> > @dricazenck - Just a thought - Instead of making sections static, would it be better if we keep them variable so it can be expanded it future easily? I don't know how the frontend team plans to use the response but if we return sections as array of objects then they can simply loop through the sections array and don't really need to know each name of the section (for the object key e.g. **HomeSection**, **ProgrammesSection**).
> > **tag** can be camelCase or snake_case as per the convention u r following.
> > `{ "title" : "Home page", "tag" : "home_page", "description": "Description of the page", "totalSections": 1, "sections" : [ { "title" : "Home section", "tag" : "home_section", "description" : "Description of the section", "images": [ { "path": "something", "alt": "something", "type": "desktop" } ] } ] }`
> > image
> 
> Not sure if I get what you mean, but the FE needs to know the name so we apply the correct components/setup for it 🤔 not all sections are the same, and if we add more the FE also needs to add that specific section/setup

I was thinking - the **tag** can be used to differentiate which component/config gets applied. That way the fields in the section object are always the same irrespective of which component/section it is. 
(but bear in mind, I don't know how u plan to implement the FE so it may be easier if its component name matches the key name).

File: api_home.yml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+          $ref: '#/components/schemas/EventsSection'
+        volunteerSection:
+          $ref: '#/components/schemas/VolunteerSection'
+
+    HeroSection:
+      type: object
+      properties:
+        title:
+          type: string
+          example: Women Coding Community
```
Comment (dricazenck) [inline api_home.yml:69]: @kikilondon for example here, in the section `Hero` there are title, description and images. If you check there are `example` for title and subtitle. So when you check the output of the api with swagger automatically. But these are only examples, of contract. The implementation will be done in the backend.

**Review by goelsonali — APPROVED** [review]
This PR should have had only the Home Page API changes. Not other changes related to about API.

File: resources/home.json (hunk 1 above)
Comment (dricazenck) [reply resources/home.json:20]: that was the comment from @lilyndk in the [requirements document](https://docs.google.com/document/d/1eD9i8JMZepTjKE9wbHJ3iIqS75DGqgoJe_YM7TdRMQ8/edit?disco=AAABO6RSdZI)

**Comment by dricazenck** [comment]
> This PR should have had only the Home Page API changes. Not other changes related to about API.

Actually, we discussed with @stepsen89 and she asked to have one api per page, which mean frontend does not have to make 6 calls to backend to get the complete page. That is why is a single page.

**Comment by dricazenck** [comment]
> > This PR should have had only the Home Page API changes. Not other changes related to about API.
> 
> @goelsonali Actually, we discussed with @stepsen89 and she asked to have one api per page, which mean frontend does not have to make 6 calls to backend to get the complete page. That is why is a single page.

For example to update then yes, we are going to have one per session, because has to be simple one for sure. Let's align on that today during the meeting. Thank you for checking.

**Comment by dricazenck** [comment]
> > @dricazenck - Just a thought - Instead of making sections static, would it be better if we keep them variable so it can be expanded it future easily? I don't know how the frontend team plans to use the response but if we return sections as array of objects then they can simply loop through the sections array and don't really need to know each name of the section (for the object key e.g. **HomeSection**, **ProgrammesSection**).
> > **tag** can be camelCase or snake_case as per the convention u r following.
> > `{ "title" : "Home page", "tag" : "home_page", "description": "Description of the page", "totalSections": 1, "sections" : [ { "title" : "Home section", "tag" : "home_section", "description" : "Description of the section", "images": [ { "path": "something", "alt": "something", "type": "desktop" } ] } ] }`
> > image
> 
> Not sure if I get what you mean, but the FE needs to know the name so we apply the correct components/setup for it 🤔 not all sections are the same, and if we add more the FE also needs to add that specific section/setup

@stepsen89 Let's clarify today during the meeting about this PR, because looks like we both have different understanding from our previous talks. Also Thanks  @mads6879 for the suggestion and to bring this topic, it is clear we have to align that.

File: resources/home.json (hunk 1 above)
Comment (dricazenck) [inline resources/home.json:20]:
```suggestion
        "icon": "diversity_2"
```

**Pull request merged**